host/src/environment.rs

```diff
@@ -0,0 +1,18 @@
+use crate::{wasi_environment, WasiCtx};
+
+#[async_trait::async_trait]
+impl wasi_environment::WasiEnvironment for WasiCtx {
+    async fn get(&mut self) -> anyhow::Result<Vec<(String, String)>> {
+        self.env
+            .iter()
+            .map(|s| {
+                let (k, v) = s.split_at(
+                    s.find('=')
+                        .ok_or_else(|| anyhow::anyhow!("expected to find '=' in {s}"))?,
```

Comment:
Could this perhaps be stored, at rest, as `Vec<(String, String)>`?

Reply:
Yeah, I was trying to minimize the changes needed to `wasi-common`, but I agree this is worth doing.

Reply:
BTW, @pchickey is `StringArray` worth keeping around?  If I switch `WasiCtx::env` to a `Vec<(String, String)>` as @alexcrichton suggests, we'll lose the size checks `StringArray` provides, and I don't know what the original intent of those checks was.  I can also work on generic-izing `StringArray` if that's worthwhile.

Reply:
StringArray was really just specific to witx, lets get rid of it!